agent/testagent.go

```diff
@@ -165,15 +185,15 @@ func (a *TestAgent) Start(t *testing.T) *TestAgent {
 			a.Agent = agent
 			break
 		} else if i == 0 {
-			require.Fail("%s %s Error starting agent: %s", id, a.Name, err)
+			return fmt.Errorf("%s %s Error starting agent: %s", id, a.Name, err)
 		} else if a.ExpectConfigError {
-			// Panic the error since this can be caught if needed. Pretty gross way to
-			// detect errors but enough for now and this is a tiny edge case that I'd
-			// otherwise not have a way to test at all...
-			panic(err)
+			return fmt.Errorf("a.ExpectConfigError: %v", err)
```

Comment:
I'm a bit confused by the rationale for why a `panic` was being called here. Is a test somewhere relying on recovering from this panic?

Reply:
good question. when i deleted it, my intention was to just see if it broke any tests. i haven't finished debugging the tests yet, so i'll leave this here and get back to you.

Reply:
alright I figured out where this was being used. was going to refactor the code to fix the use of panic here, but just wrote a comment instead.